File: Mod/src/main/java/random/cml/entity/Element.java
package random.cml.entity;

public class Element {

    private static int length = 0;
    public static final Element LIGHTNING = new Element();
    public static final Element WATER = new Element();
    public static final Element FIRE = new Element();
    public static final Element ICE = new Element();
    public static final Element EARTH = new Element();
    public static final Element AIR = new Element();
    public static final Element LIGHT = new Element();
    public static final Element DARK = new Element();
    private int flag;
    public int id;

    public Element() {
        this.flag = length++;
        this.id = 1 << flag;
    }
}

File: Battles/CML/Battles/Element.cs
using System;

namespace CML.Battles
{
    public enum Element
    {
        Lightning,
        Water,
        Fire,
        Ice,
        Earth,
        Air,
        Light,
        Dark
    }

    public static class Elements
    {
        private static readonly Element[] ElementValues = (Element[]) Enum.GetValues(typeof(Element));

        public static Element[] Values()
        {
            return ElementValues;
        }
        
        public static bool HasAdvantage(this Element current, Element other)
        {
            return other == Element.Lightning && (int) current == Values().Length - 1 || (int) current == (int) other - 1;
        }
    }
}
